### ISSUE TITLE
Feature Request: Use "Quantity unit for prices" as reference in Purchase and Inventory

### ISSUE THREAD

**Issue by HRandt** [grocy/grocy#2587]
In "Purchase" and "Inventory" sections, when price is informed, the reference doesn't use the "Quantity unit for prices" but the "Quantity unit for inventory" (see number 1 in image). This is very confusing to me. I don't want to change the "Quantity unit for inventory", as I need that reference to know if I entered the correct quantity (see number 2 in image).

It would be nice to have a way to use "Quantity unit for prices" for this prices reference. Or maybe to be able to assign a new Quantity unit for both references.

![image](https://github.com/user-attachments/assets/8070befb-715b-488d-969e-053c40c97082)

![2024-10-02 12_30_10-Edit product _ Grocy y 9 páginas más - Trabajo_ Microsoft​ Edge](https://github.com/user-attachments/assets/c5dc9c8c-6550-4018-98a9-2cd4e24e5783)

**Comment by berrnd** [comment]
> "Quantity unit for inventory"

Just quickly clarifying terminology, since often this leads to confusion:

The related product option the mentioned price-hint currently refers to is called "Quantity unit stock" ([example](https://demo-prerelease.grocy.info/product/5#qu_id_stock)), not "Quantity unit ~for inventory~".

**Comment by HRandt** [comment]
Yes, right! Sorry, English is not my first language and I got confuse with the spanish translation (Unidad de inventario). I meant "Quantity unit stock" all the time. I will correct the first post.

**Comment by gianlubiscotto** [comment]
I'm facing the same issue! To me it's useful to see the stock in grams but to check the prices in kg. Having the price hint referring to grams cause the system to have 0 USD per gram and consequently to have the item at 0 USD / kg.

Please consider to change this otherwise we need to keep the stock unit to kg!

**Comment by berrnd** [comment]
> Please consider to change

This request is still open, so it's accepted to be done tomorrow or tomorrow in 100 years. Remember that this is hobby project, so driven by desire and spare time only, spamming around doesn't make things be done faster, it's in fact more the opposite. Jumping in and implementing it right tonight yourself is also an option thanks to that this all here is Open Source from A to Z.

&nbsp;
> otherwise we need to keep the stock unit to kg

Since you can still use any unit the product's QU stock has a direct or indirect relation to everywhere (we just talked about that in that other thread), I don't get the big problem.
